src/main/java/org/codehaus/mojo/license/AbstractDownloadLicensesMojo.java

```diff
@@ -715,26 +730,57 @@ private void downloadLicenses( ProjectLicenseInfo depProject )
                 if ( !quiet )
                 {
                     getLog().warn( "POM for dependency " + depProject.toString() + " has an invalid license URL: "
-                                       + license.getUrl() );
+                                       + licenseUrl );
                 }
             }
             catch ( FileNotFoundException e )
             {
                 if ( !quiet )
                 {
                     getLog().warn( "POM for dependency " + depProject.toString()
-                                       + " has a license URL that returns file not found: " + license.getUrl() );
+                                       + " has a license URL that returns file not found: " + licenseUrl );
                 }
             }
             catch ( IOException e )
             {
                 getLog().warn( "Unable to retrieve license for dependency: " + depProject.toString() );
-                getLog().warn( license.getUrl() );
+                getLog().warn( licenseUrl );
                 getLog().warn( e.getMessage() );
             }
 
         }
 
     }
 
+    private String rewriteLicenseUrlIfNecessary( final String originalLicenseUrl )
+    {
+        String resultUrl = originalLicenseUrl;
+        if ( licenseUrlReplacements != null )
+        {
+            for ( LicenseUrlReplacement licenseUrlReplacement : licenseUrlReplacements )
+            {
+                String regexp = licenseUrlReplacement.getRegexp();
+                String replacement = licenseUrlReplacement.getReplacement();
+
+                if ( regexp != null && replacement != null )
```

Comment:
Are you sure `<replacement></replacement>` produces an empty string rather than `null`?

Reply:
> It seems that Maven treats an empty elements as if it were absent and gives the plugin a null.

If so, are substring deletions possible with the present code? Say I want to remove a trailing slash from `http://example.com/`. Would the following work? You say replacement is `null` in that case, so should you not interpret it as an empty string?

```
<licenseUrlRegexpReplacements>
  <licenseUrlRegexpReplacement>
     <regexp>/$</regexp>
     <replacement></replacement>
  </licenseUrlRegexpReplacement>
<licenseUrlRegexpReplacements>
```

Reply:
Good spot, I agree with you.  I am now interpreting null as an empty string. Javadoc updated.